### PR TITLE
Fix #2852: add optional 'property' parameter to getVehicleHandling function

### LINKED ISSUE #2852: Add optional 'property' argument to getVehicleHandling
### Is your feature request related to a problem? Please describe.

There should be an easier way to get a particular handling value, instead of the entire table

### Describe the solution you'd like

Add `property` argument to getVehicleHandling function.

### PR DESCRIPTION
**Pull request by XJMLN**
Fixes #2852 

Added optional 'property' argument to get particular vehicle handling value.

Usage: 
 `getVehicleHandling(vehicle, "mass")`

### CODE REVIEW

File: Server/mods/deathmatch/logic/luadefs/CLuaHandlingDefs.cpp [hunk 1]
```diff
@@ -390,160 +390,214 @@ int CLuaHandlingDefs::SetModelHandling(lua_State* luaVM)
...
-            lua_pushstring(luaVM, "diesel");
-        else if (eEngineType == CHandlingEntry::ELECTRIC)
-            lua_pushstring(luaVM, "electric");
+        if (argStream.NextIsString())
+        {
+            SString strProperty;
+            argStream.ReadString(strProperty);
+
+            eHandlingProperty eProperty = m_pHandlingManager->GetPropertyEnumFromName(strProperty);
+            if (eProperty)
```
Comment (Pirulax) [inline Server/mods/deathmatch/logic/luadefs/CLuaHandlingDefs.cpp]: Please invert this, and do an early out instead

**Comment by botder** [comment]
I don't understand why the code (handling in general) has to "guess" what type each property is. All we need is an array for each entry in `enum eHandlingProperty` to tell you what the underlying type for the property is, because the type never changes.

File: Server/mods/deathmatch/logic/luadefs/CLuaHandlingDefs.cpp [hunk 2]
```diff
@@ -390,14 +390,70 @@ int CLuaHandlingDefs::SetModelHandling(lua_State* luaVM)
...
 
     if (!argStream.HasErrors())
     {
+        if (argStream.NextIsString())
+        {
+            SString strProperty;
+            argStream.ReadString(strProperty);
+
+            eHandlingProperty eProperty = m_pHandlingManager->GetPropertyEnumFromName(strProperty);
+            if (!eProperty)
```
Comment (tederis) [inline Server/mods/deathmatch/logic/luadefs/CLuaHandlingDefs.cpp]: This won't work. Should be `eProperty == HANDLING_MAX`.

File: Client/mods/deathmatch/logic/luadefs/CLuaVehicleDefs.cpp [hunk 3]
```diff
@@ -2641,12 +2641,69 @@ int CLuaVehicleDefs::SetVehicleHandling(lua_State* luaVM)
...
 
     if (!argStream.HasErrors())
     {
+        if (argStream.NextIsString())
+        {
+            SString strProperty;
+            argStream.ReadString(strProperty);
+
+            eHandlingProperty eProperty = g_pGame->GetHandlingManager()->GetPropertyEnumFromName(strProperty);
+            if (!eProperty)
```
Comment (tederis) [inline Client/mods/deathmatch/logic/luadefs/CLuaVehicleDefs.cpp]: This won't work. Should be `eProperty == HANDLING_MAX`.

**Comment by Dutchman101** [comment]
Thanks!

**Pull request merged**